### PR TITLE
Consolidate unlock vault functionality for auth into a single helper method

### PR DESCRIPTION
**Pull request by david-livefront**
## 🎟️ Tracking

N/A

## 📔 Objective

This PR cleans up the logic around unlocking the vault from the `AuthRepository` placing everything into a single helper method.

## ⏰ Reminders before review

- Contributor guidelines followed
- All formatters and local linters executed and passed
- Written new unit and / or integration tests where applicable
- Used internationalization (i18n) for all UI strings
- CI builds passed
- Communicated to DevOps any deployment requirements
- Updated any necessary documentation or informed the documentation team

## 🦮 Reviewer guidelines

- 👍 (`:+1:`) or similar for great changes
- 📝 (`:memo:`) or ℹ️ (`:information_source:`) for notes or general info
- ❓ (`:question:`) for questions
- 🤔 (`:thinking:`) or 💭 (`:thought_balloon:`) for more open inquiry that's not quite a confirmed
  issue and could potentially benefit from discussion
- 🎨 (`:art:`) for suggestions / improvements
- ❌ (`:x:`) or ⚠️ (`:warning:`) for more significant problems or concerns needing attention
- 🌱 (`:seedling:`) or ♻️ (`:recycle:`) for future improvements or indications of technical debt
- ⛏ (`:pick:`) for minor or nitpick changes

### CODE REVIEW

File: app/src/main/java/com/x8bit/bitwarden/data/auth/repository/AuthRepositoryImpl.kt [hunk 1]
```diff
@@ -1377,17 +1373,15 @@ class AuthRepositoryImpl(
             if (isDeviceUnlockAvailable) {
                 unlockVaultWithTdeOnLoginSuccess(
```
Comment (brian-livefront) [inline app/src/main/java/com/x8bit/bitwarden/data/auth/repository/AuthRepositoryImpl.kt:1374]: Can we also update the docs for all these existing helpers for now to make clear they are unlocking the vault for the active user? So changing

```kotlin
    /**
     * Attempt to unlock the vault with trusted device specific data.
     */
    private suspend fun unlockVaultWithTdeOnLoginSuccess(
        loginResponse: GetTokenResponseJson.Success,
        userStateJson: UserStateJson,
        deviceData: DeviceDataModel?,
    ): VaultUnlockResult? {
```

to something like

```kotlin
    /**
     * Attempt to unlock the vault for the current user with trusted device specific data.
     */
```

or

```kotlin
    /**
     * Attempt to unlock the current user's vault with trusted device specific data.
     */
```

File: app/src/main/java/com/x8bit/bitwarden/data/auth/repository/AuthRepositoryImpl.kt [hunk 2]
```diff
@@ -1631,6 +1605,29 @@ class AuthRepositoryImpl(
...
+        return vaultRepository.unlockVault(
+            userId = userId,
+            email = accountProfile.email,
+            kdf = accountProfile.toSdkParams(),
+            privateKey = privateKey,
+            initUserCryptoMethod = initUserCryptoMethod,
+            // The value for the organization keys here will typically be null. We can separately
+            // unlock the vault for organization data after receiving the sync response if this
+            // data is currently absent. These keys may be present during certain multi-phase login
+            // processes.
```
Comment (brian-livefront) [inline app/src/main/java/com/x8bit/bitwarden/data/auth/repository/AuthRepositoryImpl.kt:1626]: 👍

File: app/src/main/java/com/x8bit/bitwarden/data/vault/repository/VaultRepository.kt [hunk 3]
```diff
@@ -188,23 +187,6 @@ interface VaultRepository : CipherManager, VaultLockManager {
...
-    @Suppress("LongParameterList")
-    suspend fun unlockVault(
-        userId: String,
-        masterPassword: String,
-        email: String,
-        kdf: Kdf,
-        userKey: String,
-        privateKey: String,
-        organizationKeys: Map<String, String>?,
-    ): VaultUnlockResult
```
Comment (brian-livefront) [inline app/src/main/java/com/x8bit/bitwarden/data/vault/repository/VaultRepository.kt:206]: 👋

File: app/src/test/java/com/x8bit/bitwarden/data/auth/repository/AuthRepositoryTest.kt [hunk 4]
```diff
@@ -1937,94 +1954,98 @@ class AuthRepositoryTest {
...
-                authModel = IdentityTokenAuthModel.MasterPassword(
-                    username = EMAIL,
-                    password = PASSWORD_HASH,
-                ),
-                captchaToken = null,
-                uniqueAppId = UNIQUE_APP_ID,
-            )
-        }
+    fun `login two factor should return Error result when get token succeeds but unlock vault fails`() =
+        runTest {
```
Comment (brian-livefront) [inline app/src/test/java/com/x8bit/bitwarden/data/auth/repository/AuthRepositoryTest.kt:1958]: Is this purely a formatting change?

Comment (david-livefront) [reply app/src/test/java/com/x8bit/bitwarden/data/auth/repository/AuthRepositoryTest.kt:1958]: Yea, I ran the auto-formatter here.

**Review by brian-livefront — DISMISSED** [review]

**Review by brian-livefront — APPROVED** [review]

**Comment by david-livefront** [comment]
Thanks @brian-livefront

**Pull request merged**